main.js

```diff
@@ -5,54 +5,77 @@ const { spawn } = require('child_process')
  * @returns {Function} handler to pass to chokidar
  */
-const createHardresetHandler = (eXecutable, hardResetMethod, argv) =>
-  () => {
-    // Detaching child is useful when in Windows to let child
-    // live after the parent is killed
-    const args = (argv || []).concat([appPath])
-    const child = spawn(eXecutable, args, {
-      detached: true,
-      stdio: 'inherit'
-    })
-    child.unref()
-    // Kamikaze!
+const createHardresetHandler = (eXecutable, hardResetMethod, argv) => () => {
+  // Detaching child is useful when in Windows to let child
+  // live after the parent is killed
+  const args = [appPath].concat(argv || [])
+  const child = spawn(eXecutable, args, {
+    detached: true,
+    stdio: 'inherit'
+  })
+  child.unref()
+  // Kamikaze!
 
-    // In cases where an app overrides the default closing or quiting actions
-    // firing an `app.quit()` may not actually quit the app. In these cases
-    // you can use `app.exit()` to gracefully close the app.
-    if (hardResetMethod === 'exit') {
-      app.exit()
-    } else {
-      app.quit()
-    }
+  // In cases where an app overrides the default closing or quiting actions
+  // firing an `app.quit()` may not actually quit the app. In these cases
+  // you can use `app.exit()` to gracefully close the app.
+  if (hardResetMethod === 'exit') {
+    app.exit()
+  } else {
+    app.quit()
   }
+}
 
+/**
+ * @typedef {Object} ExtraOptions
+ * @property {string} [hardResetMethod]
+ * @property {string[]} [argv]
+ * @property {boolean} [forceHardReset]
+ * @property {string} [electron]
+ * @property {string} [mainFile]
+ */
+
+/**
+ * @param {string | string[]} glob
+ * @param { Partial<chokidar.WatchOptions & ExtraOptions> } [options]
+ * @returns {void}
+ */
 module.exports = (glob, options = {}) => {
+  // Main file poses a special case, as its changes are
+  // only effective when the process is restarted (hard reset)
+  // We assume that electron-reload is required by the main
+  // file of the electron application if a path is not provided
+  const mainFile =
+    options.mainFile || module.parent ? module.parent.filename || '' : ''
   const browserWindows = []
-  const watcher = chokidar.watch(glob, Object.assign({ ignored: [ignoredPaths, mainFile] }, options))
+  const watcher = chokidar.watch(
+    glob,
+    Object.assign({ ignored: [ignoredPaths, mainFile] }, options)
+  )
 
   // Callback function to be executed:
   // I) soft reset: reload browser windows
-  const softResetHandler = () => browserWindows.forEach(bw => bw.webContents.reloadIgnoringCache())
+  const softResetHandler = () =>
+    browserWindows.forEach(bw => bw.webContents.reloadIgnoringCache())
   // II) hard reset: restart the whole electron process
   const eXecutable = options.electron
-  const hardResetHandler = createHardresetHandler(eXecutable, options.hardResetMethod, options.argv)
+  const hardResetHandler = createHardresetHandler(
+    eXecutable || '',
```

Comment:
I don't suppose that setting a default value here is a good idea: we should set default values once and avoid having `||` all over the code.

Reply:
Added default values and removed `||`s